contentcuration/contentcuration/frontend/channelEdit/views/ImportFromChannels/SearchOrBrowseWindow.vue

```diff
@@ -564,6 +707,125 @@
           ...this.otherRecommendations.slice(0, limit),
         ];
       },
+      initialRecommendationsEvent(request, response) {
+        const nodeToRank = {};
+        const recommendedNodes = response || [];
+        recommendedNodes.forEach(node => {
+          nodeToRank[node.node_id] = node.rank;
+        });
+
+        const content = [...this.recommendations, ...this.otherRecommendations];
+        this.recommendationsEvent = new RecommendationsEvent({
+          context: { request },
+          contentnode_id: this.importDestinationFolder.id,
+          content_id: this.importDestinationFolder.content_id,
+          target_channel_id: this.importDestinationFolder.channel_id,
+          user_id: this.userId,
+          content: content.map(node => ({
+            content_id: node.content_id,
+            node_id: node.id,
+            channel_id: node.channel_id,
+            rank: nodeToRank[node.id],
+          })),
+        });
+      },
+      handleNotRelevantRecommendation(node) {
+        const interactionEvent = new RecommendationsInteractionEvent({
+          recommendation_event_id: this.recommendationsEvent.id,
+          contentnode_id: node.id,
+          content_id: node.content_id,
+          context: {
+            //ToDo: Add appropriate context to be sent with the interaction event
+          },
+          feedback_type: FeedbackTypeOptions.rejected,
+          feedback_reason: this.recommendationsFeedback,
+        });
+        sendRequest(interactionEvent)
+          .then(() => {
+            this.rejectedNodeIds.push(node.id);
+            this.showSnackbar({
+              text: this.feedbackConfirmationMessage$(),
+              actionText: this.giveFeedbackText$(),
+              actionCallback: () => (this.showFeedbackModal = true),
+            });
+          })
+          .catch(error => {
+            this.showSnackbar({ text: this.feedbackFailedMessage$() });
+            throw error;
+          });
+      },
+      isFeedbackReasonSelected(value) {
+        return this.feedbackReason.includes(value);
+      },
+      handleFeedbackCheckboxChange(value, isChecked) {
+        if (isChecked) {
+          if (!this.feedbackReason.includes(value)) {
+            this.feedbackReason.push(value);
+          }
+        } else {
+          this.feedbackReason = this.feedbackReason.filter(item => item !== value);
+        }
+      },
+      submitRejectedRecommendationFeedback() {
+        const rejectedEvent = new RecommendationsInteractionEvent({
+          recommendation_event_id: this.recommendationsEvent.id,
+          feedback_reason: this.recommendationsFeedback,
+        });
+        if (this.validateFeedbackForm()) {
+          sendRequest(rejectedEvent, 'patch')
```

Comment:
Qn: would we still need to do a patch for this?

Reply:
Ahh I just saw that this is updating an already existing request! My apologies. I missed the patch part. I think this should be good. I was curious if there are any reasons to do this in a two step process instead of just creating one request at the end of the whole rejection event?